src/Core/Tools/SendFeatures/Services/SendValidationService.cs

```diff
@@ -0,0 +1,178 @@
+
+        return valid;
+    }
+    public async Task ValidateUserCanSaveAsync(Guid? userId, Send send)
+    {
+        if (_featureService.IsEnabled(FeatureFlagKeys.PolicyRequirements))
+        {
+            await ValidateUserCanSaveAsync_vNext(userId, send);
+            return;
+        }
+
+        if (!userId.HasValue || (!_currentContext.Organizations?.Any() ?? true))
+        {
+            return;
+        }
+
+        var anyDisableSendPolicies = await _policyService.AnyPoliciesApplicableToUserAsync(userId.Value,
+            PolicyType.DisableSend);
+        if (anyDisableSendPolicies)
+        {
+            throw new BadRequestException("Due to an Enterprise Policy, you are only able to delete an existing Send.");
+        }
+
+        if (send.HideEmail.GetValueOrDefault())
+        {
+            var sendOptionsPolicies = await _policyService.GetPoliciesApplicableToUserAsync(userId.Value, PolicyType.SendOptions);
+            if (sendOptionsPolicies.Any(p => CoreHelpers.LoadClassFromJsonData<SendOptionsPolicyData>(p.PolicyData)?.DisableHideEmail ?? false))
+            {
+                throw new BadRequestException("Due to an Enterprise Policy, you are not allowed to hide your email address from recipients when creating or editing a Send.");
+            }
+        }
+    }
+    public async Task ValidateUserCanSaveAsync_vNext(Guid? userId, Send send)
+    {
+        if (!userId.HasValue)
+        {
+            return;
+        }
+
+        var sendOptionsRequirement = await _policyRequirementQuery.GetAsync<SendOptionsPolicyRequirement>(userId.Value);
+        if (sendOptionsRequirement.DisableHideEmail && send.HideEmail.GetValueOrDefault())
+        {
+            throw new BadRequestException("Due to an Enterprise Policy, you are not allowed to hide your email address from recipients when creating or editing a Send.");
+        }
+    }
+    public async Task<long> StorageRemainingForSendAsync(Send send)
+    {
+        var storageBytesRemaining = 0L;
+        if (send.UserId.HasValue)
+        {
+            var user = await _userRepository.GetByIdAsync(send.UserId.Value);
+            if (!await _userService.CanAccessPremium(user))
+            {
+                throw new BadRequestException("You must have premium status to use file Sends.");
+            }
+
+            if (!user.EmailVerified)
+            {
+                throw new BadRequestException("You must confirm your email to use file Sends.");
+            }
+
+            if (user.Premium)
+            {
+                storageBytesRemaining = user.StorageBytesRemaining();
+            }
+            else
+            {
+                // Users that get access to file storage/premium from their organization get the default
+                // 1 GB max storage.
+                storageBytesRemaining = user.StorageBytesRemaining(
+                    _globalSettings.SelfHosted ? (short)10240 : (short)1);
```

Comment:
🎨 I'd extract the ternary into its own variable here.

```suggestion
                short limit = _globalSettings.SelfHosted ? (short)10240 : (short)1;
                storageBytesRemaining = user.StorageBytesRemaining(limit);
```